extensions/amp-recaptcha-input/0.1/test/test-amp-recaptcha-service.js

```diff
@@ -144,6 +144,31 @@ describes.realWin('amp-recaptcha-service', {
         });
   });
 
+  it('should postmessage to the specificed ' +
+    'recaptcha frame origin on execute', () => {
+
+    expect(recaptchaService.registeredElementCount_).to.be.equal(0);
+
+    const postMessageStub = sandbox.stub(
+        recaptchaService,
+        'postMessageToIframe_'
+    );
+
+    return recaptchaService
+        .register(fakeSitekey).then(() => {
+          expect(recaptchaService.unlisteners_.length).to.be.equal(3);
+
+          recaptchaService.execute(0, '');
+          recaptchaService.recaptchaApiReady_.resolve();
```

Comment:
how about post a message to simulate api ready? instead of using internal field here. that way more code is covered by the test as well.

Reply:
Is there something set up for the tests do this? For this test, we'd have to post message to the window itself, instead of an iframe. 🤔 

Also, from what I can tell, the only other thing that would be tested is that the listeners are set up correctly. Which is already covered by another test. 😄 

Let me know what you think 👍

Reply:
thinking twice, postMessage might be difficult here because we also check source origin.